tests/check/check_signed_video_sign.c

```diff
@@ -934,6 +907,49 @@ START_TEST(signing_mulitslice_stream_partial_gops)
 }
 END_TEST
 
+/**
+ * This test ensures that a reconstructed private key can be used to sign a stream again.
+ * It sets a private key, retrieves it, verifies its size, and then uses it in a new instance
+ * to sign and validate a stream.
+ */
+
+START_TEST(sign_with_reconstructed_private_key)
+{
+  SignedVideoCodec codec = settings[_i].codec;
+
+  // Create a signed video
+  signed_video_t *sv = signed_video_create(codec);
+  ck_assert(sv);
+
+  char *private_key = NULL;
+  size_t private_key_size = 0;
+
+  // Read and set the private key
+  ck_assert(read_test_private_key(settings[_i].ec_key, &private_key, &private_key_size, false));
+  signed_video_set_private_key(sv, private_key, private_key_size);
```

Comment:
Is it possible to reconstruct the private key without create->set->free of signed_video_t?

Also, and maybe more important, have you verified that it works on a device? If it does not you have a lot of dead code we need to rewrite.